File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        curr = 0
        left = 0
        summ = 0
        for i in range(k):
            summ += nums[i]
        curr = summ
        ans = curr
        for i in range(k, len(nums)):
            summ += nums[i] - nums[i-k]
            
            curr = summ
            ans = max(ans, curr)
        return ans / k
            
            
        

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int n = nums.length;
        
        // Calculate the sum of the first `k` elements
        int sum = 0;
        for (int i = 0; i < k; i++) {
            sum += nums[i];
        }

        int maxSum = sum;

        // Slide the window through the array
        for (int i = k; i < n; i++) {
            sum = sum - nums[i - k] + nums[i];
            maxSum = Math.max(maxSum, sum);
        }

        // Return the maximum average
        return (double) maxSum / k;
    }
}
